### DISCUSSION TITLE
Command line utility to display structured information (like fzf but for more structured objects)

### DISCUSSION THREAD

**Discussion by sweetbbak** [charmbracelet/inspo#31] [Ideas]
I love tools like Gum and Fzf, they are super helpful and have a million different uses. What I would love to see is a command line utility that has the ability to display information in a menu style that would let you pick an item from a list, sort of like fzf but with more than one field per menu item.

You can look at projects like Ytfzf that sort of make this work with fzf, where you have the title, creator, description and thumbnail. It would be nice to dynamically display all of this information on the command line without having to use certain hacks!

Lets say you wanted to create a makeshift music player or youtube client, you could use jq to make a json object of artist, song title, album and cover art (either manually, through bash, or even the find command) pipe that into the program and be able to display that into a menu where you can pick a song while viewing all of the information on multiple lines. You could repurpose that for really anything. I'd like to use it for my book/audio-book collection where I would be able to view more information

In a perfect world you would be able to run a command on the index of the object that is currently selected so if you wanted to view the cover art you could use whatever your terminal is capable of like icat, chafa, viu, ueberzug etc. You could play a preview of the song or something similar. After letting the user select an item, pass that object to stdin so you could feed it to mpv or another program.

At its core its just a tool to turn a JSON, CSV/TSV or otherwise, into a dynamic menu. Essentially you would create a json with objects like below, and that would be a single menu item.

`{
  "title": "A Generic Title Here",
  "status": " 11 Chapters Every 0.9 Day(s) 326 Readers 1 Reviews 03-16-2023",
  "chapters": " 11 Chapters",
  "rating": "4.3",
  "link": "https://www.novelupdates.com/series/blah/",
  "image": "https://cdn.novelupdates.com/imgmid/noimagemid.jpg"
}`

It would be a much better way to display information like this
![image](https://user-images.githubusercontent.com/75046310/226155447-27197946-84d7-4e03-90d5-da34ffea3616.png)

but in a format that is closer to this
![image](https://user-images.githubusercontent.com/75046310/226155489-8780cb8b-6c35-4f16-bb94-57e16cc39c43.png)

If no one is really interested, some basic guidance would also be appreciated. Thank you!

**Comment by pomdtr** [comment]
I think I just spent the last few months building the tool you are looking for: https://github.com/pomdtr/sunbeam

It evens use the fzf algorithm behind the scenes to filter items.

Please take a look at it!

**Reply by sweetbbak** [reply]
Wow, that's a really cool project! I'm definitely going to play around with it a lot today and see what I can get it to do.
